test/Interpreter/error_self_conformance.swift

```diff
@@ -0,0 +1,75 @@
+// RUN: %target-run-simple-swift | %FileCheck %s
+// REQUIRES: executable_test
+
+struct BoxedError<T: Error> {
+  var contents: T
+}
+
+enum MyError: Error {
+  case nothingImportant
+  case reallyImportant(String)
+}
+
+// CHECK: start
+print("start")
+
+let value = MyError.reallyImportant("hello")
```

Comment:
This is not really part of Jon's patch, it's just a prerequisite (that's already been merged).  Also, this is a test.

Reply:
On that note, can we rebase this PR?

Reply:
`git rebase -i HEAD~6` would do it, I think.

Reply:
1. `git pull --rebase upstream master` (assuming your "upstream" remote is <https://github.com/apple/swift.git>).

2. `git push origin +add-result` for a forced update (assuming your "origin" remote is <https://github.com/jshier/swift.git>).

Reply:
If you also want to squash your commits locally (in case it makes `git cherry-pick` easier).

* On macOS, if you don't like the default editor, try adding `export EDITOR=nano` to your `~/.bash_profile` (and then open a new Terminal window).

* `git rebase --interactive HEAD~5` to edit the last 5 commits.

* Except for the first line, change `pick` to `fixup` (or `squash` to keep messages). For example (SHAs may differ from yours):

  ```
  pick c5fb8e97b1 Add update Result type to standard library.
  fixup f997eb4550 Updates for review comments.
  fixup 0a2762bdf9 Update for revised proposal.
  fixup 961f3e43b2 Update test/stdlib/Result.swift
  fixup a2742e7d70 Update test/stdlib/Result.swift
  ```

* Save and exit (nano: control O, return, control X).

* `git push origin +add-result` to force the update again.

Reply:
Thanks, I think I got the rebase to work. I'm going to make some updates and then I'll see about squashing.